app/ts/components/Transfer.tsx

```diff
@@ -1,177 +1,194 @@
+
+	const showWarn = useComputed<{ id: boolean, recipient: boolean, contract: boolean, amount: boolean }>(() => {
+		const id = idInput.value !== '' && itemId.value === undefined
+		const amount = transferAmountInput.value !== '' && transferAmount.value === undefined
+		const recipient = recipientInput.value !== '' && recipientAddress.value === undefined
+		const contract = contractAddressInput.value !== '' && contractAddress.value === undefined
+		return { id, amount, recipient, contract }
+	})
 
 	async function attemptToFetchNft(address: string | undefined, id: bigint | undefined) {
 		if (address === undefined || id === undefined) {
 			return fetchingStates.value = 'empty'
 		}
 		selectedNft.value = undefined;
 		if (!provider.value?.provider) {
-			return fetchingStates.value = 'noprovider'
+			return warning.value = 'noprovider'
 		}
 		fetchingStates.value = 'fetching'
 		try {
-			const identifiedAddress = await itentifyAddress(address, id, provider.value?.provider)
+			const identifiedAddress = await itentifyAddress(address, id, provider.value?.provider, provider.value?.walletAddress)
 			if (identifiedAddress.address === contractAddress.value && identifiedAddress.inputId === itemId.value) {
 				if (identifiedAddress.type === 'ERC721') {
 					selectedNft.value = identifiedAddress
-				} else {
-					fetchingStates.value = identifiedAddress.type
+				} else if (identifiedAddress.type === 'ERC1155') {
+					selectedNft.value = identifiedAddress
+				} else if (identifiedAddress.type === 'ERC20') {
+					warning.value = 'Token address provided is an ERC20 contract'
 				}
 			}
 		} catch (e) {
+			console.log(e)
 			if (typeof e === 'object' && e !== null && 'message' in e) {
-				if (e.message === 'No ERC721 found at address') return fetchingStates.value = 'notfound'
-				if (e.message === 'Token ID does not exist') return fetchingStates.value = 'badid'
+				if (e.message === 'No ERC721 found at address') return warning.value = 'Invalid token ID for the provided ERC721'
+				if (e.message === 'Token ID does not exist') return warning.value = 'badid'
 			}
 			return console.error(e)
 		}
 	}
 
 	function sendTransfer() {
-		if (selectedNft.value && recipient.value && provider.value) {
-			transferNft(selectedNft.value, recipient.value, provider.value.provider)
-		}
+		if (selectedNft.value && recipientAddress.value && provider.value && selectedNft.value.type === 'ERC721') transferERC721(selectedNft.value, recipientAddress.value, provider.value.provider)
+		if (selectedNft.value && recipientAddress.value && provider.value && transferAmount.value && selectedNft.value.type === 'ERC1155') transferERC1155(selectedNft.value, provider.value.walletAddress, recipientAddress.value, transferAmount.value, provider.value.provider)
 	}
 
 	return (
 		<div className='flex flex-col gap-4 w-full max-w-screen-xl'>
-			<TokenPicker show={showTokenPicker} nft={selectedNft} />
+			{/* <TokenPicker show={showTokenPicker} nft={selectedNft} /> */}
```

Comment:
old code?

Reply:
Yeah, currently unneeded unless we can index balances. Removed